internal/zinc-classpath/src/main/scala/sbt/internal/inc/classpath/ClasspathUtilities.scala

```diff
@@ -74,11 +76,16 @@ object ClasspathUtilities {
   private[sbt] def filterByClasspath(classpath: Seq[File], loader: ClassLoader): ClassLoader =
     new ClasspathFilter(loader, xsbtiLoader, classpath.toSet)
 
+  /**
+   * Creates a ClassLoader that contains the classpath and the scala-library from
+   * the given instance.
+   */
   def makeLoader(classpath: Seq[File], instance: ScalaInstance): ClassLoader =
-    filterByClasspath(classpath, makeLoader(classpath, instance.loader, instance))
+    filterByClasspath(classpath, makeLoader(classpath, instance.loaderLibraryOnly, instance))
```

Comment:
We actually should make even the addition of `loaderLibraryOnly` conditional on `autoScalaLibrary := true`. In the Scala build, we have that setting disabled (because the scala-library we want on the classpath is actually a regular sub-project in our build, not the one in our reference Scala version).

Here's an example of how we get the unwanted class on our classpath: https://github.com/scala/scala/pull/6300#issuecomment-372495182

Reply:
ok. That sounds like a bug on sbt's Run or TestFramework. By the time someone called `makeLoader(...)` with `instance` I think it's ok to include `loaderLibraryOnly`. If `autoScalaLibrary` is `false` then we have to be careful not to create `ScalaInstance` based loader basically.